### ISSUE TITLE
Dissect.DPAPI

### ISSUE THREAD

**Issue by cobyge** [fox-it/dissect.target#274]
Hi,
I am interesting in creating a Dissect DPAPI module, which could be used for decrypting system secrets encrypted with DPAPI. (It could also be used to decrypt user secrets, given a plaintext password).
Reasons this would be useful:
* Parsing Wireguard configurations
* Parsing system private keys (For certificates and such)
* Decrypting artifacts that were encrypted on disk with DPAPI

I would imagine this as a separate module (named `dissect.dpapi`), which would include support for decrypting system secrets, and I would also add some plugins in `dissect.target` that would use this. (Although I could also incorporate everything into a plugin under `dissect.target`)

Before I start, I was hoping to make sure that the project owners would be willing to incorporate this.

Let me know what you think!

**Comment by Schamper** [comment]
Hi @cobyge, thank you for reaching out! DPAPI has been on my wish/todo-list for a long time now, so we're more than happy to accept your contributions!

Normally we try to limit creating new `dissect.*` packages to prevent a sprawl of packages, and decide based on things like code complexity/size and how valuable the package could be independently if we create a new package. 

I have yet to look into DPAPI, so I find it hard to estimate the amount of code necessary. Perhaps it's better to initially try to make this available under something like `dissect.target.helpers.dpapi` or `dissect.target.plugins.os.windows.dpapi` and see how much code it ends up being? We can then discuss in the MR if it being directly in `dissect.target` is the best way forward or if it indeed should be a separate package.

Let me know what your thoughts are on that, or if there are specific arguments as to why you think it's better directly served as a new package.

**Comment by cobyge** [comment]
I believe that the DPAPI package will be useful independently, but for now I'll work on adding it into `dissect.target.plugins.os.windows.dpapi`, and we'll see where we end up.

**Comment by cobyge** [comment]
I started working on implementing this, and I have a question related to use of external projects.
As part of the Masterkey File used for decrypting DPAPI secrets, there are a few enum values that effect which cryptography algorithm is used.
I found that a library by the name of dpapick (or dpapick3 for python3)  has implemented code which converts between the enum and the cryptography algorithm itself.

Is including code from external projects (GPLv3 Licensed) allowed?
I'm specifically talking about [this](https://github.com/tijldeneut/DPAPIck3/blob/main/dpapick3/crypto.py).

If not, I can implement it myself
